### DISCUSSION TITLE
How to search for the plot match to the corresponding novel

### DISCUSSION THREAD

**Discussion by File-z-J** [milvus-io/milvus#27866] [Q&A and General discussion]
If I have a number of novels, and I only remember that there is a plot (such as tens of thousands of words or tens of thousands of plots), I now enter the outline of this plot, want to search for the corresponding novel. Is it achievable? If so, could you share your thoughts?

**Comment by yhmo** [comment]
My proposal:
- for each novel, use a tokenization tool/lib to divide the large quantity of text into smaller parts. each novel might have tens of thousands of parts
- use a text embedding model to convert the smaller parts to embeddings
- store these embeddings into a vector database(Milvus or some others), along with the novel name/attributes. If you have hundreds of novels, there might be millions of embeddings in a collection of milvus.
- when you want to search a plot, convert this plot to embedding by the same embedding model
- search the embedding in the collection, find out the topk results. the novel you want to find maybe in the top1~top5 items

**Reply by File-z-J** [reply]
Hello, I have also thought about the method you mentioned, and I have the following questions:

1. There are more than 20,000 novels out there, and the list is only going to grow. Is this method suitable?

2. What I want is not an exact text match but to find a specific plot. Is there any way?

**Reply by yhmo** [reply]
There are some models that support embedding thousands of tokens, which can help to reduce the number of embeddings. Maybe you can get 10m embeddings from the 20,000 novels. Milvus can handle growing data well. I suppose the number of novels increases slowly, maybe a couple of novels every day, it is not a big problem. Just make sure the hardware resource can hold the 10m embeddings for search.
It is not an exact text match. You put a plot to search, maybe the desired novel is not in the top1~top10, you need to use some other filtering conditions to find out it.

**Reply by File-z-J** [reply]
Ok, I will put your reference into practice. Thank you very much
